File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length;
        int n = matrix[0].length;
        int low = 0, high = (m * n)-1;

        while(low <= high){
            int mid = (low + high) / 2;
            int row = mid / n;
            int col = mid % n;
            int mid_value = matrix[row][col];
            
            if(mid_value == target){
                return true;
            }
            if(mid_value > target){
                high = mid-1;
            }
            if(mid_value < target){
                low = mid+1;
            }
        }
        return false;
    }
}

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        m = len(matrix)
        n = len(matrix[0])
        low = 0
        high = (m * n)-1


        while low <= high:
            mid = (low + high) / 2
            row = mid / n
            col = mid % n
            mid_val = matrix[row][col]

            if mid_val == target:
                return True
            if mid_val < target:
                low = mid+1
            if mid_val > target:
                high = mid - 1
        return False

        
        
        